### DISCUSSION TITLE
Saving data row-by-row

### DISCUSSION THREAD

**Discussion by ontinglo** [jspsych/jsPsych#1902] [Q&A]
hi all, I am just wondering if I could do something on jsPsych code so that I could save all the data from an individual participant in one row. by doing this, it will make much easier to import the .csv file to other data analysis packages like SPSS.

Thank you so much

**Comment by cslongman** [comment]
You could probably do this, but you would need to do all of your data processing and calculate your means (etc.) in jsPsych (javascript) before exporting the single-row data file. This feels like a sub-optimal way of doing things because you would never get to see the raw data in all its glory.

If you want to do this, you woudl just need to create your data file (in whatever format you prefer) in an on_finish part of one of the final timeline nodes in your script.

Hope this helps,

Cai

**Reply by kernel52** [reply]
Hi Cai, 

would you have some starting point to do this? 
have you identified a way to change the DataCollection structure? 

I need to export a custom CSV with the data in a form that the library does not match (unfortunately :/ ) 

So far, I have only found that you can exclude some data using: save_trial_parameters: {}, but there are still some info that are registered per trial. I am exploring the .ignore() to see if I can limit the data I want to have in the output, but for the structure I haven't found yet any solution. 

thanks for any tips! 
Nikolas

**Reply by cslongman** [reply]
Hi Nikolas,

Well, I usually end up calcuating mean RT and accuracy for block-end feedback using something like what you are suggesting. It might take a bit of working out, but you are heading down the correct path.

Essentially, you just want to end up with a single-row array of data and export it as a csv.

I am only at my desk intermittently for the next two weeks, but can share some relevant code when I get back.

In the meantime, my own code is based on this...

https://github.com/fredvbrug/STOP-IT/blob/master/jsPsych_version/experiment.html

The relevant lines for you are 570-586.

Hope this helps,

Cai

**Reply by kernel52** [reply]
cool! thanks a lot for the tip, I will read it and hopefully will be able to make something out of it!

thanks a lot for your help :)

**Reply by ontinglo** [reply]
Dear Cai

thank you so much for your generous help 

highly appreciated
